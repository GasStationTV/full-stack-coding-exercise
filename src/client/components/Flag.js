import React from 'react';
import { Panel,Row,Col,Button} from 'react-bootstrap';
import PropTypes from 'prop-types';

const Flag = ({flag,openModalFunc,flagIndex,siteKey,removeAlertFunc}) => (
  <Panel>
  	<Row>Flag Type: {flag.flagType}</Row>
  	<Row>Start Date: {flag.startDate}</Row>
  	<Row>End Date: {flag.endDate}</Row>
    <Row>
      <Col sm={6} md={6} lg={6}><Button bsStyle="info" bsSize="small" onClick={() => { openModalFunc(siteKey,"EDIT",flagIndex) }}>Edit</Button></Col>
      <Col sm={6} md={6} lg={6}><Button bsStyle="danger" bsSize="small" onClick={() => { removeAlertFunc(siteKey,flagIndex) }}>Remove</Button></Col>
    </Row>
  </Panel>
);
Flag.propTypes = {
  flag: PropTypes.object,
  openModalFunc:PropTypes.func,
  flagIndex:PropTypes.number,
  siteKey:PropTypes.string,
  removeAlertFunc:PropTypes.func
};
export default Flag;
