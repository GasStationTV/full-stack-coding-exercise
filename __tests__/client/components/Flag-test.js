import React from 'react';
import Flag from '../../../src/client/components/Flag'
import renderer from 'react-test-renderer';

//Mock data
const mockFlag={startDate:"2017-01-01",endDate:"2018-01-01",flagType:"GSTV - Unsellable"}
const mockOpenModalFunc=(siteKey,addOrEditFlag,flagIndex)=>{
  console.log("Executing mockOpenModalFunc");
  console.log("siteKey: "+siteKey)
  console.log("addOrEditFlag: "+addOrEditFlag)
  console.log("flagIndex: "+flagIndex)
}
const mockFlagIndex=0
const mockSiteKey=1
const mockRemoveAlertFunc=(siteKey,flagIndex)=>{
  console.log("Executing mockRemoveAlertFunc");
  console.log("siteKey: "+siteKey)
  console.log("flagIndex: "+flagIndex)
}
const mockAddOrEditFlag="ADD"
test('Flag is renderable', () => {

  const component = renderer.create(
    <Flag flag={mockFlag} openModalFunc={mockOpenModalFunc} flagIndex={mockFlagIndex}
    siteKey={mockSiteKey} removeAlertFunc={mockRemoveAlertFunc}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
