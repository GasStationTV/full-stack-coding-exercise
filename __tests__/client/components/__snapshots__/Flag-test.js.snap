// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flag is renderable 1`] = `
<div
  className="panel panel-default"
  id={undefined}
>
  <div
    className="panel-body"
  >
    <div
      className="row"
    >
      Flag Type: 
      GSTV - Unsellable
    </div>
    <div
      className="row"
    >
      Start Date: 
      2017-01-01
    </div>
    <div
      className="row"
    >
      End Date: 
      2018-01-01
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-6 col-md-6 col-sm-6"
      >
        <button
          className="btn btn-sm btn-info"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Edit
        </button>
      </div>
      <div
        className="col-lg-6 col-md-6 col-sm-6"
      >
        <button
          className="btn btn-sm btn-danger"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Remove
        </button>
      </div>
    </div>
  </div>
</div>
`;
